ssz/tools/io.py

```diff
@@ -0,0 +1,150 @@
+            return tuple(self.parse_value(element, sedes.element_sedes) for element in value)
+
+        elif isinstance(sedes, (ByteList, ByteVector)):
+            return decode_hex(value)
+
+        elif isinstance(sedes, Container):
+            if not isinstance(value, Mapping):
+                raise ValueError(f"Expected mapping, got {type(value)}")
+            field_names_got = set(value.keys())
+            field_names_expected = set(field_name for field_name, _ in sedes.fields)
+            if field_names_got != field_names_expected:
+                raise ValueError(
+                    f"Unexpected fields: Got {field_names_got} instead of {field_names_expected}"
+                )
+            return {
+                field_name: self.parse_value(field_value, sedes.field_name_to_sedes[field_name])
+                for field_name, field_value in value.items()
+            }
+
+        elif isinstance(sedes, BaseSedes):
+            raise Exception("Unreachable: All sedes types have been checked")
+
+        else:
+            raise TypeError("Expected BaseSedes")
+
+    def dump_value(self, value, sedes=None):
+        if sedes is None or isinstance(value, BaseSerializable):
+            if not isinstance(value, BaseSerializable):
+                raise ValueError(f"Expected value is a BaseSerializable, got {type(value)}")
+            return self.dump_value(value.as_dict(), value._meta.container_sedes)
+
+        elif isinstance(sedes, Boolean):
+            if not isinstance(value, bool):
+                raise ValueError(f"Expected value of type bool, got {type(value)}")
+            return value
+
+        elif isinstance(sedes, UInt):
+            return self._dump_int(value, sedes.size)
+
+        elif isinstance(sedes, List):
+            if not isinstance(value, Sequence):
+                raise ValueError(f"Expected list, got {type(value)}")
+            return tuple(self.dump_value(element, sedes.element_sedes) for element in value)
+
+        elif isinstance(sedes, Vector):
+            if not isinstance(value, Sequence):
+                raise ValueError(f"Expected list, got {type(value)}")
+            if not len(value) == sedes.length:
+                raise ValueError(f"Expected {sedes.length} elements, got {len(value)}")
+            return tuple(self.dump_value(element, sedes.element_sedes) for element in value)
+
+        elif isinstance(sedes, (ByteList, ByteVector)):
+            return encode_hex(value)
+
+        elif isinstance(sedes, Container):
+            if not isinstance(value, Mapping):
+                raise ValueError(f"Expected mapping, got {type(value)}")
+            field_names_got = set(value.keys())
+            field_names_expected = set(field_name for field_name, _ in sedes.fields)
+            if field_names_got != field_names_expected:
+                raise ValueError(
+                    f"Unexpected fields: Got {field_names_got} instead of {field_names_expected}"
```

Comment:
You can output `field_names_got.intersection(field_names_expected)`. 😄

Reply:
Maybe `field_names_got.difference(field_names_expected)` is the desired?